frontend/app/actions/RegistrationUri.scala

```diff
@@ -0,0 +1,51 @@
+package actions
+
+import com.gu.membership.salesforce.Tier
+import com.netaporter.uri.Uri
+import configuration.Config
+import play.api.mvc.RequestHeader
+
+object RegistrationUri {
+  val REFERRER_HEADER: String = "referer"
+  val CAMPAIGN_SOURCE: String = "MEM"
+
+  def parse(request: RequestHeader) = {
+    val redirectUrl: String = Config.idWebAppRegisterUrl(request.uri)
+    extractCampaignCode(request.headers(REFERRER_HEADER), request.path) match {
+      case Some(campaignCode) => redirectUrl.concat(s"&INTCMP=${campaignCode.get}")
+      case _ => redirectUrl
+    }
+  }
+
+  private def extractCampaignCode(referer: String, path: String) = {
```

Comment:
IntelliJ tells me that the return type of this method is `Option[Some[String]]` - that's indicative of poor Scala code.

In general, you don't want to nest `Option`s - you want to flatten them into a single `Option` with flatMap, or a for-comprehension. Also, `Some` being in the actual type signature indicates that it's redundant - the return type should just be `Option[String]`